# Week 1 — Data Understanding and Basic Data Handling

## Internship Project: YouTube Trending Video Analysis — India

This notebook documents the initial stage of the internship project.

The objective of Week 1 was to understand the structure of the YouTube Trending Videos dataset and practice basic data handling operations such as loading, inspecting, filtering, sorting, and identifying data-quality issues.

In [4]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)
pd.set_option("display.float_format", lambda x: f"{x:,.2f}")

df = pd.read_csv("../data/INvideos.csv")

print("Dataset loaded successfully.")
print("Shape:", df.shape)

Dataset loaded successfully.
Shape: (37352, 16)


In [ ]:
print("Exact duplicate rows:", df.duplicated().sum())   # Check for exact number of duplicate rows 

Exact duplicate rows: 4263


In [ ]:
df.head()  

,video_id,trending_date,title,channel_title,category_id,publish_time,tags,views,likes,dislikes,comment_count,thumbnail_link,comments_disabled,ratings_disabled,video_error_or_removed,description
0,kzwfHumJyYc,17.14.11,Sharry Mann: Cute Munda ( Song Teaser) | Parmi...,Lokdhun Punjabi,1,2017-11-12T12:20:39.000Z,"sharry mann|""sharry mann new song""|""sharry man...",1096327,33966,798,882,https://i.ytimg.com/vi/kzwfHumJyYc/default.jpg,False,False,False,Presenting Sharry Mann latest Punjabi Song Cu...
1,zUZ1z7FwLc8,17.14.11,"पीरियड्स के समय, पेट पर पति करता ऐसा, देखकर दं...",HJ NEWS,25,2017-11-13T05:43:56.000Z,"पीरियड्स के समय|""पेट पर पति करता ऐसा""|""देखकर द...",590101,735,904,0,https://i.ytimg.com/vi/zUZ1z7FwLc8/default.jpg,True,False,False,"पीरियड्स के समय, पेट पर पति करता ऐसा, देखकर दं..."
2,10L1hZ9qa58,17.14.11,Stylish Star Allu Arjun @ ChaySam Wedding Rece...,TFPC,24,2017-11-12T15:48:08.000Z,Stylish Star Allu Arjun @ ChaySam Wedding Rece...,473988,2011,243,149,https://i.ytimg.com/vi/10L1hZ9qa58/default.jpg,False,False,False,Watch Stylish Star Allu Arjun @ ChaySam Weddin...
3,N1vE8iiEg64,17.14.11,Eruma Saani | Tamil vs English,Eruma Saani,23,2017-11-12T07:08:48.000Z,"Eruma Saani|""Tamil Comedy Videos""|""Films""|""Mov...",1242680,70353,1624,2684,https://i.ytimg.com/vi/N1vE8iiEg64/default.jpg,False,False,False,This video showcases the difference between pe...
4,kJzGH0PVQHQ,17.14.11,why Samantha became EMOTIONAL @ Samantha naga ...,Filmylooks,24,2017-11-13T01:14:16.000Z,"Filmylooks|""latest news""|""telugu movies""|""telu...",464015,492,293,66,https://i.ytimg.com/vi/kJzGH0PVQHQ/default.jpg,False,False,False,why Samantha became EMOTIONAL @ Samantha naga ...


In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 37352 entries, 0 to 37351
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   video_id                37352 non-null  object
 1   trending_date           37352 non-null  object
 2   title                   37352 non-null  object
 3   channel_title           37352 non-null  object
 4   category_id             37352 non-null  int64 
 5   publish_time            37352 non-null  object
 6   tags                    37352 non-null  object
 7   views                   37352 non-null  int64 
 8   likes                   37352 non-null  int64 
 9   dislikes                37352 non-null  int64 
 10  comment_count           37352 non-null  int64 
 11  thumbnail_link          37352 non-null  object
 12  comments_disabled       37352 non-null  bool  
 13  ratings_disabled        37352 non-null  bool  
 14  video_error_or_removed  37352 non-null  bool  
 15  de

In [9]:
df.columns.tolist()

['video_id',
 'trending_date',
 'title',
 'channel_title',
 'category_id',
 'publish_time',
 'tags',
 'views',
 'likes',
 'dislikes',
 'comment_count',
 'thumbnail_link',
 'comments_disabled',
 'ratings_disabled',
 'video_error_or_removed',
 'description']

In [10]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
category_id,"37,352.00",21.58,6.56,1.00,23.00,24.00,24.00,43.00
views,"37,352.00","1,060,477.65","3,184,932.05","4,024.00","123,915.50","304,586.00","799,291.25","125,432,237.00"
likes,"37,352.00","27,082.72","97,145.10",0.00,864.00,"3,069.00","13,774.25","2,912,710.00"
dislikes,"37,352.00","1,665.08","16,076.17",0.00,108.00,326.00,"1,019.25","1,545,017.00"
comment_count,"37,352.00","2,677.00","14,868.32",0.00,81.00,329.00,"1,285.00","827,755.00"


In [11]:
print("Unique video IDs:", df["video_id"].nunique())
print("Unique channels:", df["channel_title"].nunique())
print("Unique categories:", df["category_id"].nunique())
print("Unique trending dates:", df["trending_date"].nunique())

Unique video IDs: 16307
Unique channels: 1426
Unique categories: 17
Unique trending dates: 205


In [12]:
df.sort_values("views", ascending=False)[
    ["video_id", "title", "channel_title", "views", "likes", "comment_count"]
].head(10)

,video_id,title,channel_title,views,likes,comment_count
5408,FlsCjmMhFmw,YouTube Rewind: The Shape of 2017 | #YouTubeRe...,YouTube Spotlight,125432237,2912710,807558
5119,FlsCjmMhFmw,YouTube Rewind: The Shape of 2017 | #YouTubeRe...,YouTube Spotlight,113876217,2811216,787174
4936,FlsCjmMhFmw,YouTube Rewind: The Shape of 2017 | #YouTubeRe...,YouTube Spotlight,100911567,2656672,682890
4477,6ZfuNTqbHE8,Marvel Studios' Avengers: Infinity War Officia...,Marvel Entertainment,89930713,2606663,347982
4236,6ZfuNTqbHE8,Marvel Studios' Avengers: Infinity War Officia...,Marvel Entertainment,87449453,2584674,341571
4013,6ZfuNTqbHE8,Marvel Studios' Avengers: Infinity War Officia...,Marvel Entertainment,84281319,2555411,339708
3823,6ZfuNTqbHE8,Marvel Studios' Avengers: Infinity War Officia...,Marvel Entertainment,80360459,2513102,335920
4743,FlsCjmMhFmw,YouTube Rewind: The Shape of 2017 | #YouTubeRe...,YouTube Spotlight,75969469,2251815,827755
3639,6ZfuNTqbHE8,Marvel Studios' Avengers: Infinity War Officia...,Marvel Entertainment,74789251,2444956,330710
3456,6ZfuNTqbHE8,Marvel Studios' Avengers: Infinity War Officia...,Marvel Entertainment,66637636,2331352,316185


In [14]:
high_view_videos = df[df["views"] > 1_000_000]

print("Videos with more than 1 million views:", len(high_view_videos))

high_view_videos[
    ["title", "channel_title", "views", "likes", "comment_count"]
].head(10)

Videos with more than 1 million views: 7632


,title,channel_title,views,likes,comment_count
0,Sharry Mann: Cute Munda ( Song Teaser) | Parmi...,Lokdhun Punjabi,1096327,33966,882
3,Eruma Saani | Tamil vs English,Eruma Saani,1242680,70353,2684
5,"MCA (Middle Class Abbayi) TEASER - Nani,Sai Pa...",Dil Raju,6106669,98612,4763
6,Daang ( Full Video ) | Mankirt Aulakh | Sukh S...,Speed Records,5718766,127477,8063
7,Padmavati : Ek Dil Ek Jaan Video Song | Deepik...,T-Series,10588371,132738,10847
11,Jannat (Full Song) Aatish - Latest Punjabi Son...,White Hill Music,2348107,32834,1743
17,Tiger Zinda Hai | Official Trailer | Salman Kh...,YRF,35885754,829362,101117
19,The Trump Presidency: Last Week Tonight with J...,LastWeekTonight,2418783,97187,12703
20,Asli Fighter (Okka Ammayi Thappa) 2017 NEW Ful...,RKD Digital,2294242,14960,913
21,Sudigaali Sudheer Performance | Extra Jabardas...,ETV Jabardasth,1956925,9504,956


In [15]:
missing_values = df.isna().sum().sort_values(ascending=False)

missing_values[missing_values > 0]

description    561
dtype: int64

In [16]:
duplicate_count = df.duplicated().sum()

print("Exact duplicate rows:", duplicate_count)

Exact duplicate rows: 4263


## Week 1 Observations

The initial analysis identified the structure and major variables of the dataset.

The dataset contains 37,352 rows and 16 columns. Basic inspection showed that the dataset contains information about video views, likes, dislikes, comments, categories, channels, publication information and trending dates.

Sorting and filtering operations were also used to understand high-performing videos and the distribution of the available records.

Missing values and duplicate records were identified for further cleaning in Week 2.